tinyiiod.c

```diff
@@ -134,41 +146,39 @@ void tinyiiod_write_xml(struct tinyiiod *iiod)
 void tinyiiod_do_read_attr(struct tinyiiod *iiod, const char *device,
 			   const char *channel, bool ch_out, const char *attr, bool debug)
 {
-	char buf[128];
 	ssize_t ret;
 
 	if (channel)
 		ret = iiod->ops->ch_read_attr(device, channel,
-					      ch_out, attr, buf, sizeof(buf));
+					      ch_out, attr, iiod_buf[READ_BUFFER], IIOD_BUFFER_SIZE);
 	else
 		ret = iiod->ops->read_attr(device, attr,
-					   buf, sizeof(buf), debug);
+					   iiod_buf[READ_BUFFER], IIOD_BUFFER_SIZE, debug);
 
 	tinyiiod_write_value(iiod, (int32_t) ret);
 	if (ret > 0) {
-		buf[ret] = '\n';
-		tinyiiod_write(iiod, buf, (size_t) ret + 1);
+		iiod_buf[READ_BUFFER][ret] = '\n';
+		tinyiiod_write(iiod, iiod_buf[READ_BUFFER], (size_t) ret + 1);
 	}
 }
 
 void tinyiiod_do_write_attr(struct tinyiiod *iiod, const char *device,
 			    const char *channel, bool ch_out, const char *attr,
 			    size_t bytes, bool debug)
 {
-	char buf[128];
 	ssize_t ret;
 
-	if (bytes > sizeof(buf) - 1)
-		bytes = sizeof(buf) - 1;
```

Comment:
you could use here IIOD_BUFFER_SIZE, no need for sizeof.

Reply:
These lines seem to be from the previous implementation.

Reply:
This is already replaced with  IIOD_BUFFER_SIZE. Look on the right pane.

Reply:
My bad sorry.